compatibility_server/pip_checker.py

```diff
@@ -354,46 +355,79 @@ def _build_command(self, subcommands: List[str]):
         """Build pip commands."""
         return self._pip_command + subcommands
 
+    def _clone_repo(self, container: docker.models.containers.Container):
+        """Shallow clone the google-cloud-python repository."""
+        # Create a temp directory
+        _, directory = self._run_command(
+            container,
+            ['mktemp', '-d'],
+            stderr = True,
+            stdout = True,
+            raise_on_failure = True)
+
+        directory = directory.strip('\n')
+        shallow_clone_command = ['git', 'clone', '--depth', '1',
+                                 CLIENTLIBS_GITHUB_URL, directory]
+
+        self._run_command(
+            container,
+            shallow_clone_command,
+            stdout=False,
+            stderr=True,
+            raise_on_failure=True)
+        return directory
+
     def _install(self, container: docker.models.containers.Container):
         """Run pip install in the container."""
-        # If there is cloud client libraries, install it from source.
-        # Else install it from PyPI.
+        client_repo_directory = ''
+        if any(GITHUB_CLIENTLIBS_PREFIX in pkg for pkg in self._packages):
+            # The https://github.com/googleapis/google-cloud-python repository
+            # is large, has a large amount of history and contains many
+            # independent packages. Installing packages from that repository is
+            # quite slow e.g.
+            # $ time pip install git+git://...
+            # real	0m57.845s
+            # user	2m15.394s
+            # sys	0m7.130s
+            # As an optimization, if any provided package is installed from
+            # this repository, it is first cloned (using --depth=1 to avoid
+            # unnecessary history) and then installed from local disk e.g.
+            # $ time git clone --depth 1 ... && pip install ...
```

Comment:
Fill in the exact command line that you used.

Reply:
Done